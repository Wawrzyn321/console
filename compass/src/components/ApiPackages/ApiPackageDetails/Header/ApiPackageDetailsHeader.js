import React from 'react';
import LuigiClient from '@kyma-project/luigi-client';
import PropTypes from 'prop-types';

import { PageHeader, handleDelete } from 'react-shared';
import { Button } from 'fundamental-react';
import RequestInputSchemaModal from '../RequestInputSchemaModal/RequestInputSchemaModal';

import { useMutation } from '@apollo/react-hooks';
import {
  GET_API_PACKAGE,
  DELETE_API_PACKAGE,
} from 'components/ApiPackages/gql';
import { SEND_NOTIFICATION } from 'gql';

ApiPackageDetailsHeader.propTypes = {
  apiPackage: PropTypes.object.isRequired,
  application: PropTypes.object.isRequired,
};

function navigateToApplication() {
  LuigiClient.linkManager()
    .fromContext('application')
    .navigate('');
}

export default function ApiPackageDetailsHeader({ apiPackage, application }) {
  const [sendNotification] = useMutation(SEND_NOTIFICATION);

  const [deleteApiPackageMutation] = useMutation(DELETE_API_PACKAGE, {
    refetchQueries: () => [
      {
        query: GET_API_PACKAGE,
        variables: {
          applicationId: application.id,
          apiPackageId: apiPackage.id,
        },
      },
    ],
  });

  function showDeleteSuccessNotification() {
    sendNotification({
      variables: {
        content: `Deleted API Package "${apiPackage.name}".`,
        title: apiPackage.name,
        color: '#359c46',
        icon: 'accept',
        instanceName: apiPackage.name,
      },
    });
  }

  const breadcrumbItems = [
    { name: 'Applications', path: '/applications', fromContext: 'tenant' },
    { name: application.name, path: '/', fromContext: 'application' },
    {
      name: apiPackage.name,
      path: '/',
    },
    { name: '' },
  ];

  const deleteApiPackage = () => {
    handleDelete(
      'API Package',
      apiPackage.id,
      apiPackage.name,
      id => deleteApiPackageMutation({ variables: { id } }),
      () => {
        navigateToApplication();
        showDeleteSuccessNotification();
      },
    );
  };

  const deleteButton = (
    <Button onClick={deleteApiPackage} type="negative" option="light">
      Delete
    </Button>
  );

  return (
    <PageHeader
      title={apiPackage.name}
      breadcrumbItems={breadcrumbItems}
      actions={deleteButton}
    >
      <PageHeader.Column title="Name">{apiPackage.name}</PageHeader.Column>
      <PageHeader.Column title="Description" columnSpan={2}>
        {apiPackage.description}
      </PageHeader.Column>
      <PageHeader.Column title="Request Input Schema" columnSpan={3}>
        <RequestInputSchemaModal
          schema={apiPackage.instanceAuthRequestInputSchema}
        />
      </PageHeader.Column>
    </PageHeader>
  );
}
