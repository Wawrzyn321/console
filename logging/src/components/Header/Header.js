import React from 'react';
import PropTypes from 'prop-types';
import './Header.scss';

import { Panel } from 'fundamental-react';

import SearchInput from './SearchInput/SearchInput';
import LabelsInput from './LabelsInput/LabelsInput';
import AdvancedSettings from './AdvancedSettings/AdvancedSettings';
import LabelsDisplay from './LabelsDisplay/LabelsDisplay';
import BottomToolbar from './BottomToolbar/BottomToolbar';

Header.propTypes = {
  updateFilteringState: PropTypes.func.isRequired,
  searchPhrase: PropTypes.string.isRequired,
  labels: PropTypes.array.isRequired,
  sortDirection: PropTypes.oneOf(['ascending', 'descending']),
  readonlyLabels: PropTypes.array.isRequired,
  advancedSettings: PropTypes.object.isRequired,
};

export default function Header({
  updateFilteringState,
  searchPhrase,
  labels,
  sortDirection,
  readonlyLabels,
  advancedSettings,
}) {
  const [advancedShown, setAdvancedShown] = React.useState(false);

  function addLabel(label) {
    if (!labels.includes(label)) {
      updateFilteringState({ labels: [...labels, label] });
    }
  }

  function removeLabel(label) {
    updateFilteringState({ labels: labels.filter(l => l !== label) });
  }

  function removeAllLabels() {
    updateFilteringState({ labels: [] });
  }

  function toggleAdvancedSettingsVisibility() {
    setAdvancedShown(!advancedShown);
  }

  const advancedSettingsButtonText = advancedShown
    ? 'Hide Advanced Settings'
    : 'Show Advanced Settings';

  return (
    <Panel className="fd-has-padding-regular fd-has-padding-bottom-none">
      <h1 className="fd-has-type-3 fd-has-padding-bottom-tiny">Logs</h1>
      <section className="header__settings-group">
        <LabelsInput readonlyLabels={readonlyLabels} addLabel={addLabel} />
        <SearchInput
          searchPhrase={searchPhrase}
          updateFilteringState={updateFilteringState}
        />
        <span
          className="link-button fd-has-type-minus-1 header__settings-group__toggle"
          onClick={toggleAdvancedSettingsVisibility}
        >
          {advancedSettingsButtonText}
        </span>
      </section>
      {advancedShown && (
        <AdvancedSettings
          advancedSettings={advancedSettings}
          hideSettings={() => setAdvancedShown(false)}
          updateFilteringState={updateFilteringState}
        />
      )}
      <div>
        <LabelsDisplay
          labels={labels}
          readonlyLabels={readonlyLabels}
          removeLabel={removeLabel}
          removeAll={removeAllLabels}
        />
        <BottomToolbar
          sortDirection={sortDirection}
          updateFilteringState={updateFilteringState}
        />
      </div>
    </Panel>
  );
}
