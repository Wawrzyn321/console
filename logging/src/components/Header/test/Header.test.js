import React from 'react';
import renderer from 'react-test-renderer';
import Header from './../Header';

describe('Header', () => {
  console.error = jest.fn();

  afterEach(() => {
    console.error.mockReset();
  });

  afterAll(() => {
    expect(console.error.mock.calls[0][0]).toMatchSnapshot(); // unique "key" prop warning
  });

  it('Renders with minimal props', () => {
    const component = renderer.create(
      <Header
        updateFilteringState={() => {}}
        searchPhrase="search phrase"
        readonlyLabels={['a', 'b']}
        labels={['a', 'b']}
        sortDirection="ascending"
        advancedSettings={{}}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
