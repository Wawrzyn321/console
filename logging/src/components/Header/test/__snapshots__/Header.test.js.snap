// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders with minimal props 1`] = `
<div
  className="fd-panel fd-has-padding-regular fd-has-padding-bottom-none"
>
  <h1
    className="fd-has-type-3 fd-has-padding-bottom-tiny"
  >
    Logs
  </h1>
  <section
    className="header__settings-group"
  >
    <section
      className="fd-display-l-inline-block"
    >
      <span
        className="caption-muted"
      >
        Labels
      </span>
      <div
        className="fd-combobox-input"
      >
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <div
                className="fd-combobox-control"
                onClick={[Function]}
              >
                <div
                  className="fd-input-group fd-input-group--after"
                >
                  <input
                    className="fd-input"
                    placeholder="Select Label"
                    type="text"
                  />
                  <span
                    className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button"
                  >
                    <button
                      className=" fd-button--light sap-icon--navigation-down-arrow"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      className="fd-has-margin-right-small"
    >
      <span
        className="caption-muted search-input__caption"
      >
        Search
        <div
          className="small-inline-help-wrapper"
        >
          <span
            className="fd-inline-help"
          >
            <span
              className="fd-inline-help__content fd-inline-help__content--right"
            >
              Search for logs by text (optional)
            </span>
          </span>
        </div>
      </span>
      <input
        className=""
        id="search-input"
        min="0"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value="search phrase"
      />
    </section>
    <span
      className="link-button fd-has-type-minus-1 header__settings-group__toggle"
      data-test-id="advanced-settings-toggle"
      onClick={[Function]}
    >
      Show Advanced Settings
    </span>
  </section>
  <div>
    <section
      className="labels-display"
    >
      <ul>
        <li>
          <span
            className="fd-token caption-muted y-fd-token--no-button"
            role="button"
          >
            a
          </span>
        </li>
        <li>
          <span
            className="fd-token caption-muted y-fd-token--no-button"
            role="button"
          >
            b
          </span>
        </li>
        <li>
          <span
            className="fd-token caption-muted"
            onClick={[Function]}
            role="button"
          >
            a
          </span>
        </li>
        <li>
          <span
            className="fd-token caption-muted"
            onClick={[Function]}
            role="button"
          >
            b
          </span>
        </li>
      </ul>
      <span
        className="link-button fd-has-type-minus-1 fd-has-margin-left-tiny"
        data-test-id="clear-all"
        onClick={[Function]}
      >
        Clear All
      </span>
    </section>
    <div
      className="header__options-wrapper"
    >
      <button
        className="fd-button--light sap-icon--media-play link-button fd-has-margin-right-tiny fd-has-type-minus-1"
        data-test-id="auto-refresh-button"
        onClick={[Function]}
        size="xs"
      >
        auto refresh
      </button>
      <span
        className="link-button fd-has-type-minus-1 select-dropdown"
      >
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <button
                className="fd-button--light sap-icon--past fd-has-margin-right-tiny"
                onClick={[Function]}
                size="xs"
              />
            </div>
          </div>
        </div>
      </span>
      <span
        className="link-button fd-has-type-minus-1 select-dropdown"
      >
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <button
                className="fd-button--light sap-icon--sort fd-has-margin-right-tiny"
                onClick={[Function]}
                size="xs"
              >
                ascending
              </button>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
</div>
`;
